### PR TITLE
Bluetooth: host: Add PHY update procedure options

### LINKED ISSUE #25165: LE Coded Phy code rate switch [s2/s8]
**Is your enhancement proposal related to a problem? Please describe.**
 #24618 allow change PHY mode, but look like it's impossible switch code rate. 

**Describe the solution you'd like**
I want function that add ability to select code rate.

### PR DESCRIPTION
**Pull request by joerchan**
Add options for phy update procedure. User can now set no preference
option for a particular PHY as well as preference for LE Coded PHY
coding scheme.

Fixes: #25165

### CODE REVIEW

File: include/bluetooth/conn.h [hunk 1]
```diff
@@ -81,10 +81,29 @@ struct bt_conn_le_phy_info {
...
+	/** Convenience value when no options are specified. */
+	BT_CONN_LE_PHY_OPT_NONE = 0,
+
+	/** LE Coded using S=2 coding preferred when transmitting. */
+	BT_CONN_LE_PHY_OPT_CODED_S2  = BIT(0),
+
+	/** LE Coded using S=8 coding preferred when transmitting. */
+	BT_CONN_LE_PHY_OPT_CODED_S8  = BIT(1),
+
+	/** No preference for transmit PHY, pref_rx_phy is ignored. */
```
Comment (rugeGerritsen) [inline include/bluetooth/conn.h:95]: Should we rather let `pref_tx_phy = 0` indicate that there is no preference for transmit PHY? Similar for RX.

**Comment by jhedberg** [comment]
@joerchan @carlescufi this PR is marked as fixing #23989 , however that issue is already closed. Should the reference be removed from this PR?

**Comment by joerchan** [comment]
> @joerchan @carlescufi this PR is marked as fixing #23989 , however that issue is already closed. Should the reference be removed from this PR?

Sorry, wrong PHY issue. Fixed now.

File: include/bluetooth/conn.h (hunk 1 above)
Comment (joerchan) [reply include/bluetooth/conn.h:95]: @carlescufi @jhedberg Any thoughts on removing the two "no preference" options, and instead have `BT_CONN_LE_PHY_OPT_NONE = 0`, mean no preference? I'm a bit indicisive, I think i like this suggestion better.

Comment (jhedberg) [reply include/bluetooth/conn.h:95]: none == no preference sounds intuitive and simpler to me

**Review by rugeGerritsen — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**